go/vt/wrangler/materializer.go

```diff
@@ -727,12 +729,11 @@ func (wr *Wrangler) ExternalizeVindex(ctx context.Context, qualifiedVindexName s
 	if sourceVindex == nil {
 		return fmt.Errorf("vindex %s not found in vschema", qualifiedVindexName)
 	}
-	qualifiedTableName := sourceVindex.Params["table"]
-	splits = strings.Split(qualifiedTableName, ".")
-	if len(splits) != 2 {
-		return fmt.Errorf("table name in vindex should be of the form keyspace.table: %s", qualifiedTableName)
+
+	targetKeyspace, targetTableName, err := sqlparser.ParseTable(sourceVindex.Params["table"])
+	if err != nil || targetKeyspace == "" {
+		return fmt.Errorf("table name in vindex should be of the form keyspace.table: %s", sourceVindex.Params["table"])
```

Comment:
Again, nit and not something you added but we should make this consistent with the error message from above:
```
fmt.Errorf("vindex table name must be in the form <keyspace>.<table>. Got: %v", sourceVindex.Params["table"])
```

We could alternatively add a format string and use that instead to ensure uniformity...
```
tableNameErrStrf = `vindex table name must be in the form <keyspace>.<table>. Got: %v`
...
fmt.Errorf(tableNameErrStrf, sourceVindex.Params["table"])
```

I don't feel strongly either way, unless this is used in even more places within the file. :-)

Reply:
Also fixed :smile:.